tests/test_routes.py

```diff
@@ -148,8 +149,35 @@ def test_update_recommendation(self):
             json=new_recommendation,
             content_type="application/json"
         )
-       #new_recommendation["relationship"] = "CROSS_SELL"
-       #resp = self.app.put("/recommendations")
         self.assertEqual(resp.status_code, status.HTTP_200_OK)
         updated_recommendation = resp.get_json()
-        self.assertEqual(updated_recommendation["relationship"], "CROSS_SELL")
\ No newline at end of file
+        self.assertEqual(updated_recommendation["relationship"], "CROSS_SELL")
+
+    def test_update_recommendation_not_found(self):
+        """ update a Recommendation that is not found """
+        test_recommendation = RecommendationFactory()
+        test_recommendation.product_id1 = 0
+        test_recommendation.product_id2 = 0
+        logging.debug(test_recommendation)
+        resp = self.app.put("/recommendations/products/0/related-products/0",json=test_recommendation.serialize(),
+            content_type="application/json")
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_update_recommendation_no_content_type(self):
+        """ create a Recommendation with no content type """
+        resp = self.app.put("/recommendations/products/0/related-products/0")
+        self.assertEqual(resp.status_code, status.HTTP_415_UNSUPPORTED_MEDIA_TYPE)
+
+    def test_update_recommendation_no_relation_type(self):
+        """ create a Recommendation with no content type """
+        test_recommendation = RecommendationFactory()
+        test = json.dumps({'product_id1': test_recommendation.product_id1, "product_id2": test_recommendation.product_id2, "relationship": " "})
+        logging.debug(test_recommendation.serialize())
+        logging.debug(test)
+        resp = self.app.put(
+            "/recommendations/products/{}/related-products/{}".format(test_recommendation.product_id1,
+                                                                      test_recommendation.product_id2),
+            json=test,
+            content_type="application/json"
+        )
+        self.assertEqual(resp.status_code, status.HTTP_415_UNSUPPORTED_MEDIA_TYPE)
```

Comment:
Ok I see your error. So it's actually the deserialization error meaning that input relationship cannot be empty at all, in which case you could remove this test and remove ```if not recommendation``` because it won't be able to hit here at all because validation error would raise first before it could reach this check. 
``
DataValidationError: Invalid YourResourceModel: body of request contained bad or no data
   -------------------- >> begin captured logging << --------------------
   root: DEBUG: {'product_id1': 3, 'product_id2': 8, 'relationship': 'CROSS_SELL'}
   root: DEBUG: {"product_id1": 3, "product_id2": 8, "relationship": " "}
   --------------------- >> end captured logging << ---------------------
```

Reply:
Yeah. I test this in test_model part. "if not recommendation" is implement in model.py